openfe/utils/system_probe.py

```diff
@@ -273,6 +273,55 @@ def _get_hostname() -> str:
     return socket.gethostname()
 
 
+def _slurm_environment() -> bool:
+    """
+    Check if the current environment is managed by SLURM.
+    """
+    
+    slurm_job_id = os.environ.get("SLURM_JOB_ID")
+    
+    if slurm_job_id:
+        return True
+    else:
+        return False
+
+
+def _check_slurm_gpu_info() -> bool:
+    """
+    Check if the GPU information is available in the SLURM environment.
+
+    Returns
+    -------
+    bool
+        True if the GPU information is available in the SLURM environment, False
+        otherwise.
+
+    Notes
+    -----
+    This function checks if the GPU information is available in the SLURM environment by
+    inspecting the environment variables.
+
+    The function returns True if any of the following environment variables are present:
+    - 'SLURM_JOB_GPUS'
+    - 'SLURM_GPUS'
```

Comment:
I'm afraid these might be very specific to some deployments of SLURM (at least I know in the clusters I've managed in the past we didn't have these env variables set).

Reply:
Thanks for the feedback, appreciate it!  :D
I applied this changes on my end to test it with CPU. But if using CPU for production is a bad idea in itself then it is not worth it to merge in the main branch.

Reply:
Not sure this is mentioned in the openfe docs - maybe it would be worth to include a small comment about it if it's not.